### DISCUSSION TITLE
Equivalent to Apache's RewriteCond %{HTTP:Accept-Language} / RewriteRule wanted

### DISCUSSION THREAD

**Discussion by personwithproxy** [dotnet/yarp#2130] [Q&A] [answered]
We're replacing Apache with YARP and searching for a YARP equivalent to the following Apache directives:

RewriteCond %{HTTP:Accept-Language} ^(ja|pt-BR).*$ [NC]
RewriteRule ^/svc/help(.*)$ http://localhost:5080/svc/help/%1$1 [P,NC,L]

Is this possible with JSON configuration only (I could not find a solution) or must be done programmatically?
YARP 2.0/.NET 6

Thank you!

**Comment by MihaZupan** [comment, marked as answer]
The JSON config doesn't let you capture values from headers to use in the path right now.

If you only wanted to do this for a few languages, you can try something like this, where you define a separate route depending on the `Accept-Language` header value:
```json
"Routes": {
  "catchAll": {
    "ClusterId": "myCluster",
    "Match": {
      "Path": "{**catch-all}"
    }
  },
  "help-ja": {
    "ClusterId": "myCluster",
    "Match": {
      "Path": "/svc/help/{**remainder}",
      "Headers": [
        { "Name": "Accept-Language", "Values": [ "ja" ], "Mode": "Contains" }
      ]
    },
    "Transforms": [
      { "PathPattern": "/svc/help/ja/{**remainder}" }
    ]
  },
  "help-ptBR": {
    "ClusterId": "myCluster",
    "Match": {
      "Path": "/svc/help/{**remainder}",
      "Headers": [
        { "Name": "Accept-Language", "Values": [ "pt-BR" ], "Mode": "Contains" }
      ]
    },
    "Transforms": [
      { "PathPattern": "/svc/help/pt-BR/{**remainder}" }
    ]
  }
}
```

but it may end up being easier to do the transformation via code.

**Reply by personwithproxy** [reply]
Nice approach, works like a charm, thank you!
